push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java

```diff
@@ -209,4 +210,43 @@ public void messageSendFailed(ApnsNotification message, Throwable e) {
         // null if, why ever, there was no cert/passphrase
         return null;
     }
+
+    /**
+     * Configure the Gateway to the Apns servers.
+     * Default gateway and port can be override with respectively :
+     *  - aerogear.apns.push.host
+     *  - aerogear.apns.push.port
+     *
+     * Feedback gateway and port can be override with  respectively :
+     *  - aerogear.apns.feedback.host
+     *  - aerogear.apns.feedback.port
+     * @param iOSVariant
+     * @param builder
+     */
+    private void configureDestinations(iOSVariant iOSVariant, ApnsServiceBuilder builder) {
+        // pick the destination:
+        if (iOSVariant.isProduction()) {
+            builder.withProductionDestination();
+        } else {
+            builder.withSandboxDestination();
+        }
+
+        //Is the gateway host&port provided by a system property ?
+        if(tryGetProperty(CUSTOM_AEROGEAR_APNS_PUSH_HOST)!= null){
+            int port = Utilities.SANDBOX_GATEWAY_PORT;
+            if(tryGetProperty(CUSTOM_AEROGEAR_APNS_PUSH_PORT)!= null) {
+                port = Integer.parseInt(tryGetProperty(CUSTOM_AEROGEAR_APNS_PUSH_PORT));
+            }
+            builder.withGatewayDestination(tryGetProperty(CUSTOM_AEROGEAR_APNS_PUSH_HOST),port);
+        }
+
+        //Is the feedback gateway provided by a system property ?
+        if(tryGetProperty(CUSTOM_AEROGEAR_APNS_FEEDBACK_HOST)!= null){
```

Comment:
still, this is executed all the time. As said earlier, let's move that into a static block - we really need that info once